src/backend/joanie/core/api/client.py

```diff
@@ -477,6 +479,52 @@ def validate(
         order.validate()
         return Response(status=200)
 
+    @action(
+        detail=True,
+        methods=["GET"],
+    )
+    def get_contract(
+        self, request, pk=None
+    ):  # pylint: disable=no-self-use, unused-argument, invalid-name
+        """
+        Return the PDF in bytes of the contract from an order where the product has a contract
+        definition.
+        """
+        order = self.get_object()
+
+        if not order.product.contract_definition:
+            raise ValidationError("No contract definition attached to the product")
+
+        context_data = order.product.contract_definition.get_document_context(order)
+        contract_definition_pdf_bytes = issuers.generate_document(
+            "contract_definition", context_data
+        )
+        response = HttpResponse(
+            contract_definition_pdf_bytes, content_type="application/pdf", status=200
+        )
+        response[
+            "Content-Disposition"
+        ] = f"attachment; filename={order.product.contract_definition.title}.pdf;"
+
+        return response
+
+    @action(detail=True, methods=["POST"])
+    def submit_for_signature(
+        self, request, pk=None
+    ):  # pylint: disable=no-self-use, unused-argument, invalid-name
+        """
+        Create the contract of a product's order that has a contract definition and it submits
+        the contract to the signature provider. It returns a one-time use invitation link.
+        """
+        order = self.get_object()
+
+        if not self.request.user == order.owner:
+            raise PermissionDenied("The current user is not the order's owner.")
+
+        invitation_link = order.submit_for_signature()
+
+        return Response(invitation_link, status=200)
```

Comment:
I think we should put it in a json object

Reply:
Indeed, since it's an invitation link that we return. thanks